lib/UserDefinedAbstract.php

```diff
@@ -0,0 +1,601 @@
+	 * timestamp as integer.
+	 * Can be easily overwritten for example if a special format has to be expected on the
+	 * source website.
+	 * @param string $value
+	 * @return string
+	 */
+	protected function formatItemContent($value) {
+		return $value;
+	}
+
+	/**
+	 * Formats the URI of a feed item. Takes extracted raw URI and returns it formatted
+	 * as string.
+	 * Can be easily overwritten for in case the value needs to be transformed into something
+	 * else.
+	 * @param string $value
+	 * @return string
+	 */
+	protected function formatItemUri($value) {
+		if(strlen($value) === 0) {
+			return '';
+		}
+		if(strpos($value, 'http://') === 0 || strpos($value, 'https://') === 0) {
+			return $value;
+		}
+
+		return urljoin($this->feedUri, $value);
+	}
+
+	/**
+	 * Formats the author of a feed item. Takes extracted raw author and returns it formatted
+	 * as string.
+	 * Can be easily overwritten for in case the value needs to be transformed into something
+	 * else.
+	 * @param string $value
+	 * @return string
+	 */
+	protected function formatItemAuthor($value) {
+		return $value;
+	}
+
+	/**
+	 * Formats the timestamp of a feed item. Takes extracted raw timestamp and returns unix
+	 * timestamp as integer.
+	 * Can be easily overwritten for example if a special format has to be expected on the
+	 * source website.
+	 * @param string $value
+	 * @return false|int
+	 */
+	protected function formatItemTimestamp($value) {
+		return strtotime($value);
+	}
+
+	/**
+	 * Formats the enclosures of a feed item. Takes extracted raw enclosures and returns them
+	 * formatted as array.
+	 * Can be easily overwritten for in case the values need to be transformed into something
+	 * else.
+	 * @param $value
+	 * @return array
+	 */
+	protected function formatItemEnclosures($value) {
+		return $value;
+	}
+
+	/**
+	 * Formats the categories of a feed item. Takes extracted raw categories and returns them
+	 * formatted as array.
+	 * Can be easily overwritten for in case the values need to be transformed into something
+	 * else.
+	 * @param $value
+	 * @return array
+	 */
+	protected function formatItemCategories($value) {
+		return $value;
+	}
+
+	/**
+	 * Formats the uid of a feed item. returns null if the input is empty.
+	 * Can be easily overwritten for in case the values need to be transformed into something
+	 * else.
+	 * @param string $value
+	 * @return null|string
+	 */
+	protected function formatItemUid($value) {
+		return empty($value) ? null : $value;
+	}
+
+	/**
+	 * Fixes feed encoding by invoking PHP's utf8_decode function on extracted texts.
+	 * Useful in case of "broken" or "weird" characters in the feed where you'd normally
+	 * expect umlauts.
+	 *
+	 * @param $input
+	 * @return string
+	 */
+	protected function fixEncoding($input) {
+		return $this->getParam('fix_encoding') ? utf8_decode($input) : $input;
+	}
+
+	/**
+	 * @param $mediaUrl
+	 * @return string|void
+	 */
+	protected function cleanMediaUrl($mediaUrl) {
```

Comment:
This method belongs somewhere else.

Reply:
This change is outside the scope of this pr.